### DISCUSSION TITLE
Issue with GH Actions: site is build, hosted, but no content

### DISCUSSION THREAD

**Discussion by skramm** [getpelican/pelican#3288] [Q&A] [answered]
Hi,
New user here, used to have a hosted Wordpress for ages, but finally realizing there was no added value for what I use it, so here I am starting with Pelican. First impressions ave very positive, great job!

I plan to use a github pages hosting, so I did carefully read these pages:
https://docs.getpelican.com/en/4.9.1/tips.html#publishing-to-github-pages-using-a-custom-github-actions-workflow

But I wanted to store in the repo only the content folder, not all the Pelican system files, to keep the repo clean.
From my knowledge of how GH actions work (I used it for other projects), GH provides a "runner" on which all the actions are build, and then the produced content is moved to the "hosting" part.
So I assumed this must be possible.

So first, I made a try: in some local folder, I ran the "quickstart":
(`$ pelican-quickstart`)

Then I moved the `content` folder up a level, added a Markdown page in `content/pages`, and did:
`$ pelican ../content --listen`

Worked fine, I see the Pelican site and page locally, so I understand that Pelican is able to build a site whose content is outside of the Pelican main folder.

Now my problem:
I have set up a repo dedicated to my content, added the `.github/workflows/pelican.yml` as given above, added a `content/pages/dummy1.md` file.
Also added at the repo top level the two configuration files `pelicanconf.py` and `publishconf.py`.

Nothing special in these, I saw that the first one has a 
`PATH = "content"` parameter/value, so I assumed that was fine.

Now, commit, push, checking on GH actions, I see green/green for "deploy/build" and "deploy/deploy".
And I do see online the Pelican site, but... **no content**, the top banner says "This site currently has no content."

I tried changing the `PATH` variable in `pelicanconf.py` to `./content` but that didn't change anything.

Any idea how I can make that work?

**Comment by skramm** [comment]
Ah... seems my problem is solved

I **did** actually read this discussion before posting: #3174 
(and the associated workflow file)
But as it was very similar to the "official" workflow file, I assumed that it had already had been merged to the main docs.
So I didn't try it before (verrrry sorry), and just did:
So: just replaced the line:
```
uses: "getpelican/pelican/.github/workflows/github_pages.yml@master"
```
by
```
uses: seanh/pelican-github-pages/.github/workflows/pelican.yml@main
```
And that magically solved the issue. My content is now visible online.

But the question could be still relevant: what is the problem with the "official" workflow GH file? Did I miss something?

**Comment by avaris** [comment]
> Did I miss something?

I'm going to guess: yes. Especially the `requirements` option description:

```
The Python requirements to install, for example to enable markdown and typogrify use: "pelican[markdown] typogrify" or if you have a requirements file: "-r requirements.txt"
```

Default github pages workflow does not install markdown and I suspect that's your issue.

**Reply by skramm** [reply, marked as answer]
Thanks for your reply.
Ok, I got it. I assumed wrongly that Markdown support was inherent to Pelican.
After re-reading, I can see that this is clearly stated:
> Markdown generation requires that you first explicitly install the [Python-Markdown](https://github.com/Python-Markdown/markdown) package, which can be done via pip install Markdown.

So, I assume that the workflow file provided by @seanh  does install this package on the GH runner, while the "official"  workflow file does not.
After checking, I can see indeed [that line in his file](https://github.com/seanh/pelican-github-pages/blob/87f1dcd4dca1c58b0496ef3e2b718eb693b247d7/.github/workflows/pelican.yml#L37)

Closing, and thanks for pointing me in the right direction.

**Reply by seanh** [reply]
Hi @skramm, to use Markdown with the official GitHub Actions workflow include `requirements: "pelican[markdown]"` in the `with:` section in your workflow file. For example:

```yml
name: Deploy to GitHub Pages
on:
  push:
    branches: ["main"]
  workflow_dispatch:
jobs:
  deploy:
    uses: "getpelican/pelican/.github/workflows/github_pages.yml@master"
    permissions:
      contents: "read"
      pages: "write"
      id-token: "write"
    with:
      settings: "publishconf.py"
      requirements: "pelican[markdown]"
```

Docs: https://docs.getpelican.com/en/latest/tips.html#publishing-to-github-pages-using-a-custom-github-actions-workflow

**Reply by skramm** [reply]
Thanks !
Actually, yours is working perfectly (but I assume with that mod they are more or less the same ! ;-) )

**Reply by seanh** [reply]
More or less, yes. I'd recommend using the official one, as I don't provide any guarantees about maintaining mine

**Reply by skramm** [reply]
Ok, got it, I'll move on, thank you.
